src/cryptonote_core/tx_pool.cpp

```diff
@@ -594,6 +594,36 @@ namespace cryptonote
     return true;
   }
   //---------------------------------------------------------------------------------
+  bool tx_memory_pool::get_complement(const std::vector<crypto::hash> &hashes, std::vector<cryptonote::blobdata> &txes) const
+  {
+    CRITICAL_REGION_LOCAL(m_transactions_lock);
+    CRITICAL_REGION_LOCAL1(m_blockchain);
+
+    m_blockchain.for_all_txpool_txes([this, &hashes, &txes](const crypto::hash &txid, const txpool_tx_meta_t &meta, const cryptonote::blobdata*) {
+      const auto i = std::find(hashes.begin(), hashes.end(), txid);
```

Comment:
Crap I missed this review, but this can be done immediately after the PR. This should be sorted and then `std::lower_bound` somewhere. Otherwise, the peer can jam a long list of bogus hashes and DoS easier.

Reply:
Or no wait, sorting is a `n*logn` operation, and this is only searching by a smaller amount in the txpool. So hmm. The stuff in txpool is probably already sorted by hash too, which would accelerate the search. But I think that's part of internal DB details.